### PR TITLE
refactor: use provider urls instead of api keys for provider initiation

### PR DESCRIPTION
**Pull request by WilliamHawkingStephen**
## What does this MR do?

- Improving the way to initialize providers by provider url instead of using api keys.
- resolve dependency of api keys 

## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/.env.example [hunk 1]
```diff
@@ -102,10 +102,11 @@ ZKSYNCV1_MIDDLE_ORDERER_PRIV_KEY=private-key
 ZKSYNCV1_GOERLI_MIDDLE_ORDERER_PRIV_KEY=private-key
 # ---------------------------------------------------------------------
 
-# Third Party API Keys And App IDs
+# Third Party API Keys - URLs And App IDs
 # ---------------------------------------------------------------------
-# Infura Project IDs(Third Party Node)
-INFURA_PROJECT_IDS=id1,id2,id3
+# Providers URLs(Third Party Node)
+TESTNET_PROVIDERS_URLS=https://goerli.infura.io/v3/YOUR_INFURA_PROJECT_ID
```
Comment (hypersolo) [inline ts-version/.env.example:108]: it should be divided by network too.

File: ts-version/src/data/blockchain/evm.ts [hunk 2]
```diff
@@ -39,19 +39,22 @@ export default class Evm implements IBlockchain {
...
   public broadcasters: buildWalletsOutput;
+
   public startPromise: Promise<Evm>;
 
   constructor(
     public network: string,
     public providerNetwork: string,
     public contractAddress: string,
-    private _apiKeys: string[],
+    // private _apiKeys: string[],
```
Comment (hypersolo) [inline ts-version/src/data/blockchain/evm.ts:49]: delete comment

File: ts-version/src/data/blockchain/evm.ts [hunk 3]
```diff
@@ -39,19 +39,22 @@ export default class Evm implements IBlockchain {
...
+    private _providerUrls: string[],
   ) {}
 
   public async start(): Promise<Evm> {
     await this.initProviders({
-      apiKeys: this._apiKeys,
+      // apiKeys: this._apiKeys,
```
Comment (hypersolo) [inline ts-version/src/data/blockchain/evm.ts:55]: delete comment

File: ts-version/src/data/blockchain/evm.ts [hunk 4]
```diff
@@ -379,12 +383,16 @@ export default class Evm implements IBlockchain {
   }
 
   public static getInstance(name: string, networkName: string): Evm {
+    const testnetProviders = ["zksyncv1_goerli", "ethereum_goerli"];
```
Comment (hypersolo) [inline ts-version/src/data/blockchain/evm.ts:386]: why do we refer to zksync in evm class?

if we set env variables like ETHEREUM_GOERLI_PROVIDERS_URLS, we just call env.get(`${this.network.toUppercase()}_PROVIDERS_URLS`) and there is no need for extra check.

**Review by hypersolo — CHANGES_REQUESTED** [review]

File: ts-version/src/data/blockchain/evm.ts (hunk 4 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/data/blockchain/evm.ts:386]: nice :+1: 
we will apply this

File: ts-version/src/data/blockchain/evm.ts (hunk 2 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/data/blockchain/evm.ts]: done

File: ts-version/src/data/blockchain/evm.ts (hunk 3 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/data/blockchain/evm.ts]: done

File: ts-version/.env.example (hunk 1 above)
Comment (WilliamHawkingStephen) [reply ts-version/.env.example]: done

**Review by hypersolo — APPROVED** [review]

**Pull request merged**